### PR TITLE
OpenFL 4 compatibility

### PR DESCRIPTION
**Pull request by Beeblerox**
second iteration of openfl 4 compatibility changes

### CODE REVIEW

File: flixel/addons/api/FlxGameJolt.hx [hunk 1]
```diff
@@ -709,7 +709,11 @@ class FlxGameJolt
 	{
 		_loader.removeEventListener(Event.COMPLETE, parseData);
 		
+		#if flash
 		if (Std.string(e.currentTarget.data) == "")
+		#else
+		if (Std.string((cast e.currentTarget).data) == "")
```
Comment (Gama11) [inline flixel/addons/api/FlxGameJolt.hx]: Why is this necessary?

File: flixel/addons/util/FlxScene.hx [hunk 2]
```diff
@@ -158,7 +158,7 @@ class FlxScene
 							addInstance(instance, container, element);
 
 						case "button":
-							var instance = new FlxButton();
+							var instance = new FlxButton(0, 0, "Label");
```
Comment (Gama11) [inline flixel/addons/util/FlxScene.hx]: This seems unrelated?

Comment (Beeblerox) [inline flixel/addons/util/FlxScene.hx]: it fixes issue when buttons created by FlxScene don't show labels

File: flixel/addons/api/FlxGameJolt.hx (hunk 1 above)
Comment (Gama11) [inline flixel/addons/api/FlxGameJolt.hx]: I figured, but I was wondering what was changed that made this necessary.

File: flixel/addons/util/FlxScene.hx (hunk 2 above)
Comment (Gama11) [inline flixel/addons/util/FlxScene.hx]: Seems like this shouldn't be necessary.

**Comment by Gama11** [comment]
See https://github.com/HaxeFlixel/flixel/pull/2032#issuecomment-301263890.
